From 4d12753a659ba3b9a3a9bd24e2f842e5bd0fa7a2 Mon Sep 17 00:00:00 2001
From: zhangxiaoyu <zhangxiaoyu58@huawei.com>
Date: Fri, 22 Jul 2022 10:01:52 +0800
Subject: [PATCH 2/5] add HAVE_ISULAD definition for lxc header

Signed-off-by: zhangxiaoyu <zhangxiaoyu58@huawei.com>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad32724..d7e668b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,6 +63,7 @@ message("--  commit id: " ${GIT_COMMIT_HASH})
 add_definitions(-DLCRPATH="${CMAKE_INSTALL_PREFIX}/var/lib/lcr")
 add_definitions(-DLOGPATH="${CMAKE_INSTALL_PREFIX}/var/log/lcr")
 add_definitions(-DLCR_GIT_COMMIT="${GIT_COMMIT_HASH}")
+add_definitions(-DHAVE_ISULAD)
 
 configure_file(
 	"${CMAKE_CURRENT_SOURCE_DIR}/config.h.in"
-- 
2.25.1

