From 88f2a60d014299241b6e53f60340b2debc1adc10 Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Tue, 3 Nov 2020 17:23:57 +0800
Subject: [PATCH 3/3] fix ctest error when use make

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 tests/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 8110f10..39d0270 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -83,7 +83,7 @@ target_link_libraries(mock_ut
     ${GTEST_LIBRARY}
     pthread
     )
-add_dependencies(mock_ut log_ut libocispec_ut)
+add_dependencies(mock_ut log_ut libocispec_ut go_crc64_ut)
 
 # run ut and gcov
 if (ENABLE_GCOV)
-- 
2.20.1

