From 00abacb3321b85dfb75039b2af7a447500f62c8b Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Thu, 16 Jun 2022 07:38:43 +0100
Subject: [PATCH 4/4] use vsnprintf to replace vsprintf

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 src/buffer.c | 2 +-
 src/error.c  | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/buffer.c b/src/buffer.c
index f9955d6..39e0360 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -194,7 +194,7 @@ int buffer_nappendf(Buffer *buf, size_t length, const char *format, ...)
     }
 
     va_start(argp, format);
-    status = vsprintf(tmp, format, argp);
+    status = vsnprintf(tmp, length, format, argp);
     va_end(argp);
     if (status < 0) {
         goto error;
diff --git a/src/error.c b/src/error.c
index bed77d7..71c4856 100644
--- a/src/error.c
+++ b/src/error.c
@@ -68,7 +68,7 @@ void lcr_set_error_message(lcr_errno_t errcode, const char *format, ...)
     va_list argp;
     va_start(argp, format);
 
-    ret = vsprintf(errbuf, format, argp);
+    ret = vsnprintf(errbuf, BUFSIZ, format, argp);
     va_end(argp);
     clear_error_message(&g_lcr_error);
     if (ret < 0) {
@@ -89,7 +89,7 @@ void lcr_try_set_error_message(lcr_errno_t errcode, const char *format, ...)
         return;
     }
     va_start(argp, format);
-    ret = vsprintf(errbuf, format, argp);
+    ret = vsnprintf(errbuf, BUFSIZ, format, argp);
     va_end(argp);
     clear_error_message(&g_lcr_error);
     if (ret < 0) {
@@ -109,7 +109,7 @@ void lcr_append_error_message(lcr_errno_t errcode, const char *format, ...)
     va_list argp;
     va_start(argp, format);
 
-    ret = vsprintf(errbuf, format, argp);
+    ret = vsnprintf(errbuf, BUFSIZ, format, argp);
     va_end(argp);
     if (ret < 0) {
         g_lcr_error.errcode = LCR_ERR_FORMAT;
-- 
2.20.1

