From 84368fe8c022bd8e0d0d8968bf9325070f677383 Mon Sep 17 00:00:00 2001
From: wujing <wujing50@huawei.com>
Date: Tue, 16 Mar 2021 09:59:37 +0800
Subject: [PATCH 05/10] add secure compile options prohibit rpath options

Signed-off-by: wujing <wujing50@huawei.com>
---
 lcr.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lcr.spec b/lcr.spec
index 71d6abd..6911cd5 100644
--- a/lcr.spec
+++ b/lcr.spec
@@ -54,7 +54,7 @@ the %{name}-libs package contains libraries for running iSula applications.
 %build
 mkdir -p build
 cd build
-%cmake -DDEBUG=ON -DENABLE_UT=ON -DLIB_INSTALL_DIR=%{_libdir} ../
+%cmake -DDEBUG=ON -DENABLE_UT=ON -DCMAKE_SKIP_RPATH=TRUE -DLIB_INSTALL_DIR=%{_libdir} ../
 %make_build
 
 %install
-- 
2.25.1

