From b6845431ea3aafc32419430bef4a3e59781d7133 Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Wed, 24 Nov 2021 07:38:39 +0000
Subject: [PATCH 6/6] support null value in json

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 third_party/libocispec/common_c.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/third_party/libocispec/common_c.py b/third_party/libocispec/common_c.py
index 78224c7..968d5d2 100644
--- a/third_party/libocispec/common_c.py
+++ b/third_party/libocispec/common_c.py
@@ -126,6 +126,7 @@ static yajl_gen_status gen_yajl_val (yajl_val obj, yajl_gen g, parser_error *err
         case yajl_t_false:
             return yajl_gen_bool (g, false);
         case yajl_t_null:
+            return yajl_gen_null(g);
         case yajl_t_any:
             return __stat;
     }
-- 
2.20.1

