From fac14fbabd07c587d09cb96ef9ccb663096dca3e Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Tue, 23 Mar 2021 10:05:04 +0800
Subject: [PATCH 06/10] remove invalid fuzz option

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 tests/fuzz/fuzz.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/fuzz/fuzz.sh b/tests/fuzz/fuzz.sh
index d71be7c..7513e65 100755
--- a/tests/fuzz/fuzz.sh
+++ b/tests/fuzz/fuzz.sh
@@ -23,7 +23,7 @@
 #!/bin/bash
 
 LIB_FUZZING_ENGINE="/lib64/libFuzzer.a"
-FUZZ_OPTION="corpus -dict=./dict/log_fuzz.dict -runs=10000000 -max_total_time=3600"
+FUZZ_OPTION="-dict=./dict/log_fuzz.dict -runs=10000000 -max_total_time=3600"
 
 if [ ! -f "$LIB_FUZZING_ENGINE" ];then
     echo "$LIB_FUZZING_ENGINE not exist, pls check"
-- 
2.25.1

