From 02054466c71822e197499b0271cfed57c96598b2 Mon Sep 17 00:00:00 2001
From: haozi007 <liuhao27@huawei.com>
Date: Thu, 11 Jun 2020 15:14:48 +0800
Subject: [PATCH 06/11] fix log test error

Signed-off-by: haozi007 <liuhao27@huawei.com>
---
 tests/log_llt.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/log_llt.cpp b/tests/log_llt.cpp
index a047f57..8944b38 100644
--- a/tests/log_llt.cpp
+++ b/tests/log_llt.cpp
@@ -42,7 +42,7 @@ TEST(log_testcases, test_isula_libutils_default_log_config)
     ASSERT_EQ(tconf.file, nullptr);
     ASSERT_EQ(tconf.driver, nullptr);
     EXPECT_STREQ(name, tconf.name);
-    EXPECT_STREQ("NOTSET", tconf.priority);
+    EXPECT_STREQ("FATAL", tconf.priority);
 
     // not quiet configs check
     tconf.quiet = false;
-- 
2.25.1

