From 312ac2de6c5b1f7ea446c968f30139149810318b Mon Sep 17 00:00:00 2001
From: wujing <wujing50@huawei.com>
Date: Thu, 9 Dec 2021 10:30:45 +0800
Subject: [PATCH] fix the error of gcc compilation optimization level

Signed-off-by: wujing <wujing50@huawei.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5222555..557ac34 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ endif()
 
 option(DEBUG "set lcr gcc option" ON)
 if (DEBUG STREQUAL "ON")
-    add_definitions("-g -o2")
+    add_definitions("-g -O2")
 endif()
 
 option(ENABLE_UT "enable ut" OFF)
-- 
2.25.1

